### PR TITLE
Enable FP16 scalars and use placeholder for alpha tensor in add/sum ops

### PR DESCRIPTION
**Pull request by razarmehr**
There are still accuracy problems with FP16 which will be addressed in future PRs.

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/BinaryOps.mm [hunk 1]
```diff
@@ -102,54 +111,60 @@ void div_mode_template(const Tensor& self, const Tensor& other,
...
-  binaryOpTensor(self, other, output, op_name + "_out_mps:" + (rounding_mode.has_value() ? c10::str(*rounding_mode) : ""), div_mode_op_block);
+  binaryOpTensor(self, other, Scalar(1.0), output, op_name + "_out_mps:" + (rounding_mode.has_value() ? c10::str(*rounding_mode) : ""), div_mode_op_block);
 }
 
 void add_sub_template(const Tensor& self, const Tensor& other, const Scalar& alpha, const Tensor& output, std::string op_name)
 {
-  BinaryOpBlock add_sub_op_block = ^BinaryOpFn() {
-    double alpha_val = alpha.toDouble();
-    MPSGraphTensor* secondaryTensor = secondary;
+  const bool alpha_has_value = alpha.toDouble() != 1.0;
```
[the remark this answers is not in the document]
Comment (razarmehr) [reply aten/src/ATen/native/mps/operations/BinaryOps.mm:123]: Done

File: test/test_mps.py [hunk 2]
```diff
@@ -3913,34 +3915,46 @@ def helper(shape, low, high, dtype=torch.int32):
...
+            for dtype in [torch.float16, torch.float32]:
+                cpu_x = torch.randn(shape, device='cpu', dtype=dtype, requires_grad=False)
+                mps_x = cpu_x.detach().clone().to('mps')
+
+                cpu_y = torch.randn(shape, device='cpu', dtype=dtype, requires_grad=False)
+                mps_y = cpu_y.detach().clone().to('mps')
+
+                cpu_out = torch.add(cpu_x, cpu_y, alpha=alpha)
+                mps_out = torch.add(mps_x, mps_y, alpha=alpha)
+                # fp16 isn't accurate when alpha is passed
```
[the remark this answers is not in the document]
Comment (razarmehr) [reply test/test_mps.py:3927]: Done

File: aten/src/ATen/native/mps/operations/BinaryOps.mm [hunk 3]
```diff
@@ -79,18 +87,19 @@ void binaryOpTensor(const Tensor& self, const Tensor& other, const Tensor& outpu
...
 }
 
 void div_mode_template(const Tensor& self, const Tensor& other,
                        c10::optional<c10::string_view> rounding_mode,
                        const Tensor& output, const string op_name)
 {
-  BinaryOpBlock div_mode_op_block = ^BinaryOpFn() {
-    MPSGraphTensor* divTensor =  [mpsGraph divisionWithPrimaryTensor:primary
-                                                     secondaryTensor:secondary
+  BinaryOpBlock div_mode_op_block = ^MPSGraphTensor* (MPSGraph* mpsGraph, BinaryOpCachedGraph* cachedGraph) {
```
[the remark this answers is not in the document]
Comment (razarmehr) [reply aten/src/ATen/native/mps/operations/BinaryOps.mm]: Done

**Review by kulinseth — APPROVED** [review]
